core/src/main/java/io/questdb/cutlass/text/ParallelCsvFileImporter.java

```diff
@@ -603,12 +603,21 @@ private void collectStub(final TextImportTask task) {
     }
 
     private void createWorkDir() {
-        removeWorkDir();
+        // First, create the work root dir, if it doesn't exist.
+        Path workDirPath = tmpPath.of(inputWorkRoot).slash$();
+        if (!ff.exists(workDirPath)) {
+            int errno = ff.mkdir(workDirPath, configuration.getMkDirMode());
+            if (errno != 0) {
+                throw CairoException.critical(errno).put("could not create import work root directory [path='").put(workDirPath).put("', errno=").put(errno).put("]");
+            }
+        }
 
-        Path workDirPath = tmpPath.of(importRoot).slash$();
+        // Next, remove and recreate the per-table sub-dir.
+        removeWorkDir();
+        workDirPath = tmpPath.of(importRoot).slash$();
         int errno = ff.mkdir(workDirPath, configuration.getMkDirMode());
```

Comment:
mkdir() does not return `errno`. This is a status code, 0 success, everything else error. `errno` is ff.errno()

Reply:
Good stop. Let me fix it in the original and new code.

Reply:
Done in 052d88ba088fd535b10c17abdfb887d8d6eef234